assets/js/modules/analytics-4/datastore/accounts.js

```diff
@@ -121,12 +143,43 @@ const baseActions = {
 
 		return { response, error };
 	},
+	/**
+	 * Finds a matching account and property.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object|null} Matching account and property on success, otherwise NULL.
+	 */
+	*findMatchedAccount() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const accounts = yield Data.commonActions.await(
+			registry
+				.__experimentalResolveSelect( MODULES_ANALYTICS_4 )
+				.getAccountSummaries()
+		);
+
+		if ( ! Array.isArray( accounts ) || accounts.length === 0 ) {
+			return null;
+		}
+
+		return yield Data.commonActions.await(
+			registry.dispatch( MODULES_ANALYTICS_4 ).findMatchedProperty()
+		);
+	},
```

Comment:
This is not how it should work. In your implementation this action returns a property object instead of an account. This action should be something like this:

```js
*findMatchedAccount() {
    const registry = yield Data.commonActions.getRegistry();
    const property = yield Data.commonActions.await(
        registry.dispatch( MODULES_ANALYTICS_4 ).findMatchedProperty()
    );
    
    if ( ! property ) {
        return null;
    }
    
    const summaries = registry.select( MODULES_ANALYTICS_4 ).getAccountSummaries(); // account summaries are already resolved
    for ( const account of summaries ) {
        if ( account.propertySummaries.find( ( { _id } ) => _id === property._id ) ) {
            return account;
        }
    }
    
    return null
}
```

Reply:
Updated ✅